src/parser/SGTypes.ts

```diff
@@ -0,0 +1,373 @@
+import { SourceNode } from 'source-map';
+import type { Range } from 'vscode-languageserver';
+import { isSGChildren, isSGField, isSGFunction, isSGInterface, isSGScript } from '../astUtils/xml';
+import type { FileReference } from '../interfaces';
+import type { SGTranspileState } from './SGTranspileState';
+
+export interface SGToken {
+    text: string;
+    range?: Range;
+}
+
+export interface SGAttribute {
+    key: SGToken;
+    value: SGToken;
+    range?: Range;
+}
+
+export class SGTag {
+
+    get id() {
+        return this.getAttribute('id');
+    }
+    set id(value: string) {
+        this.setAttribute('id', value);
+    }
+
+    constructor(
+        public tag: SGToken,
+        public attributes: SGAttribute[] = [],
+        public range?: Range
+    ) {}
+
+    getSGAttribute(name: string): SGAttribute | undefined {
+        return this.attributes.find(att => att.key.text.toLowerCase() === name);
+    }
+
+    getAttribute(name: string): string | undefined {
```

Comment:
What if we rename this to `getAttributeValue` or `getAttributeText`, and then change `getSGAttribute` to `getAttribute`?